server/pulp/server/managers/consumer/applicability.py

```diff
@@ -118,29 +120,52 @@ def regenerate_applicability_for_repos(repo_criteria):
         repo_criteria.fields = ['id']
         repo_ids = [r.repo_id for r in model.Repository.objects.find_by_criteria(repo_criteria)]
 
+        task_group_id = uuid4()
+
         for repo_id in repo_ids:
             # Find all existing applicabilities for given repo_id. Setting batch size of 5 ensures
             # the MongoDB cursor does not time out. See https://pulp.plan.io/issues/998#note-6 for
             # more details.
             existing_applicabilities = RepoProfileApplicability.get_collection().find(
-                {'repo_id': repo_id}).batch_size(5)
+                {'repo_id': repo_id})
+            counter = 0
```

Comment:
This will be much simpler using https://github.com/pulp/pulp/blob/master/server/pulp/plugins/util/misc.py#L14

15 lines of code basically becomes:

```
for batch in paginate(existing_applicabilities, 100):
  regenerate_applicability_for_repos_task.apply_async(repo_id, batch, group_id=task_group_id)
```